install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -831,19 +831,19 @@ data:
 
 {{- if or (eq $ipam "cluster-pool") (eq $ipam "cluster-pool-v2beta") }}
 {{- if .Values.ipv4.enabled }}
-  {{- if .Values.ipam.operator.clusterPoolIPv4PodCIDRList }}
-  cluster-pool-ipv4-cidr: {{ .Values.ipam.operator.clusterPoolIPv4PodCIDRList | join " " | quote }}
-  {{- else }}
-  cluster-pool-ipv4-cidr: {{ .Values.ipam.operator.clusterPoolIPv4PodCIDR | quote }}
+  {{- if hasKey .Values.ipam.operator "clusterPoolIPv4PodCIDR" }}
+    {{- /* ipam.operator.clusterPoolIPv4PodCIDR remove in v1.14, remove around v1.17 */ -}}
```

Comment:
Why do these comments say 1.17 while the other ones say 1.15?

Reply:
Because the other one refers to a user-facing value that we're deprecating and planning to remove.

This one refers to the failsafe we've been discussing. The failsafe isn't supposed to be visible to the user unless they keep using a value that we have already removed. Even then, this comment will not show up. This is only for us to remember to remove the failsafe when we reach a point when we no longer expect anyone to try to update from a version supporting `clusterPoolIPv4PodCIDR` to a version not supporting it. I picked 1.17 because when preparing for that version, supported branches should be 1.1{4,5,6} and none of them would support `clusterPoolIPv4PodCIDR`, so the failsafe should be safe to remove.

Does it answer your question?

Reply:
I changed to `remove _this failsafe_ around v1.17`